src/gebt_poc/gen_alpha_2D.cpp

```diff
@@ -0,0 +1,334 @@
+            states[i].GetGeneralizedCoordinates(), states[i].GetVelocity(),
+            states[i].GetAcceleration(), states[i].GetAlgorithmicAcceleration()};
+        log->Info("** Integrating step number " + std::to_string(i + 1) + " **\n");
+        states.emplace_back(
+            std::get<0>(this->AlphaStep(input_state, n_constraints, linearization_parameters))
+        );
+    }
+    log->Info("Time integration has completed!\n");
+    return states;
+}
+
+std::tuple<State, Kokkos::View<double*>> GeneralizedAlphaTimeIntegrator::AlphaStep(
+    const State& state, size_t n_constraints,
+    std::shared_ptr<LinearizationParameters> linearization_parameters
+) {
+    auto gen_coords = state.GetGeneralizedCoordinates();
+    auto velocity = state.GetVelocity();
+    auto acceleration = state.GetAcceleration();
+    auto algo_acceleration = state.GetAlgorithmicAcceleration();
+
+    // Define some constants that will be used in the algorithm
+    const auto h = this->time_stepper_.GetTimeStep();
+    const auto n_nodes = state.GetNumberOfNodes();
+    const auto size = velocity.size();
+
+    const auto kAlphaFLocal = kAlphaF_;
+    const auto kAlphaMLocal = kAlphaM_;
+    const auto kBetaLocal = kBeta_;
+    const auto kGammaLocal = kGamma_;
+
+    // Initialize some X_next variables to assist in updating the State
+    auto gen_coords_next =
+        Kokkos::View<double* [kNumberOfLieGroupComponents]>("gen_coords_next", n_nodes);
+    auto velocity_next =
+        Kokkos::View<double* [kNumberOfLieAlgebraComponents]>("velocity_next", n_nodes);
+    auto acceleration_next =
+        Kokkos::View<double* [kNumberOfLieAlgebraComponents]>("acceleration_next", n_nodes);
+    auto algo_acceleration_next =
+        Kokkos::View<double* [kNumberOfLieAlgebraComponents]>("algo_acceleration_next", n_nodes);
+    auto delta_gen_coords =
+        Kokkos::View<double* [kNumberOfLieAlgebraComponents]>("delta_gen_coords", n_nodes);
+    auto lagrange_mults_next = Kokkos::View<double*>("lagrange_mults_next", n_constraints);
+
+    // Loop over all nodes in the system and update the generalized coordinates, velocities,
+    // accelerations, and algorithmic accelerations
+    auto log = util::Log::Get();
+    log->Info(
+        "Performing Newton-Raphson iterations to update solution using the generalized-alpha "
+        "algorithm\n"
+    );
+
+    for (size_t node = 0; node < n_nodes; node++) {
+        // Perform the linear update part of the generalized alpha algorithm
+        // Algorithm from Table 1, Brüls, Cardona, and Arnold 2012
+        Kokkos::parallel_for(
+            size,
+            KOKKOS_LAMBDA(const size_t i) {
+                algo_acceleration_next(node, i) = (kAlphaFLocal * acceleration(node, i) -
+                                                   kAlphaMLocal * algo_acceleration(node, i)) /
+                                                  (1. - kAlphaMLocal);
+
+                delta_gen_coords(node, i) = velocity(node, i) +
```

Comment:
Why is `h` not included here, but then applied during `UpdateGeneralizedCoordinates`? I know that it's equivalent, I'd just like to understand why it's done this way.

Reply:
Allow me to get back to you on this please.

Reply:
Please disregard, I found the explanation in the paper describing the algorithm.